### PR TITLE
REF: simplify casting of scalars to array

### PR DESCRIPTION
**Pull request by jbrockmendel**
cast_scalar_to_array is only used in two places, and in each of them there is a better option available.

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -616,9 +615,9 @@ def __init__(
                 if arr.ndim != 0:
                     raise ValueError("DataFrame constructor not properly called!")
 
-                values = cast_scalar_to_array(
-                    (len(index), len(columns)), data, dtype=dtype
-                )
+                arr = np.atleast_2d(arr)
+                shape = (len(index), len(columns))
+                values = np.tile(arr, shape)
```
Comment (jorisvandenbossche) [inline pandas/core/frame.py:620]: Potentially even simpler would be `values = np.full(shape, arr)` ?

**Review by jorisvandenbossche — COMMENTED** [review]
Nice!

File: pandas/core/frame.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/frame.py:620]: good idea, will update

**Comment by jreback** [comment]
great, pls rebase and ping on green

**Comment by jbrockmendel** [comment]
ping

**Pull request merged**